### PR TITLE
DreamStudio integration

### PR DESCRIPTION
**Pull request by carson-katri**
This adds backend switching, with options "Local" and "DreamStudio" currently. In the future, other backends could be supported, such as an existing Stable Diffusion installation, HuggingFace's diffusers library, or something else.

DreamStudio setup is completely optional. However, releases will be made that only contain the dependencies necessary to run on DreamStudio (which will be much lighter weight). This will make the addon more accessible for lower-end hardware.

### CODE REVIEW

**Comment by carson-katri** [comment]
This includes a very experimental implementation of realtime viewport rendering, but it is not currently accessible in the addon. Realtime viewport is not planned for 0.0.8.

File: requirements-dreamstudio.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+stability-sdk
```
Comment (NullSenseStudio) [inline requirements-dreamstudio.txt:1]: Either need to include `realesrgan` or disable upscaling panel.

File: generator_process/intents/prompt_to_image.py [hunk 2]
```diff
@@ -196,4 +196,76 @@ def update(self, n=1):
...
+            # guidance_models: List[str] = None,
+        )
+
+        for answer in answers:
+            for artifact in answer.artifacts:
+                if artifact.finish_reason == interfaces.gooseai.generation.generation_pb2.FILTER:
+                    self.send_exception(False, "Your request activated DreamStudio's safety filter. Please modify your prompt and try again.")
+                if artifact.type == interfaces.gooseai.generation.generation_pb2.ARTIFACT_IMAGE:
+                    response = Image.open(io.BytesIO(artifact.binary))
+                    image_writer(response, seed)
```
Comment (NullSenseStudio) [inline generator_process/intents/prompt_to_image.py:268]: Iterations returns same seed for all images.
```suggestion
                    image_writer(response, artifact.seed)
```

**Review by NullSenseStudio — CHANGES_REQUESTED** [review]
Source image will discard the image if `Color Correct` is checked. Inpainting and outpainting are generating no different from the modify option.

File: .github/workflows/package-release.yml [hunk 3]
```diff
@@ -48,10 +48,64 @@ jobs:
...
+        uses: thedoctor0/zip-release@main
+        with:
+          type: zip
+          filename: dream_textures-macos-dreamstudio.zip
+          exclusions: '*.git*'
+      - name: Archive and upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: dream_textures-macos-intel-dreamstudio
+          path: dream_textures-macos-intel-dreamstudio.zip
```
[the remark this answers is not in the document]
Comment (carson-katri) [reply .github/workflows/package-release.yml:111]: Changed checks to look for contents of `stable_diffusion`, not its existence.

File: requirements-dreamstudio.txt (hunk 1 above)
Comment (carson-katri) [reply requirements-dreamstudio.txt:1]: Disabled upscaling panel when the DreamStudio backend is selected, as well as other unsupported options.

File: operators/dream_texture.py [hunk 4]
```diff
@@ -191,7 +195,12 @@ def step_progress_update(self, context):
         bpy.types.Scene.dream_textures_info = bpy.props.StringProperty(name="Info", update=step_progress_update)
         
         info("Waiting For Process")
-        generator = GeneratorProcess.shared()
+        if len(backend_options(self, context)) <= 1:
+            headless_prompt.backend = backend_options(self, context)[0]
```
Comment (NullSenseStudio) [inline operators/dream_texture.py:199]: `TypeError: bpy_struct: item.attr = val: expected a string enum, not tuple`
```suggestion
            headless_prompt.backend = backend_options(self, context)[0][0]
```

File: generator_process/__init__.py [hunk 5]
```diff
@@ -302,8 +317,9 @@ def main():
             # Couldn't track down exactly where it was hanging but it's good enough.
             from skimage import transform
```
Comment (NullSenseStudio) [inline generator_process/__init__.py:318]: Forgot to previously test with only dream studio dependencies. Scikit-image is not a part of them.
```suggestion
            if os.path.exists(absolute_path(".python_dependencies/skimage")):
                from skimage import transform
```

File: generator_process/registrar.py [hunk 6]
```diff
@@ -1,13 +1,58 @@
...
+                return True
+            case BackendTarget.STABILITY_SDK:
+                return False
+    
+    def upscaling(self):
+        match self:
+            case BackendTarget.LOCAL:
+                return True
+            case BackendTarget.STABILITY_SDK:
+                return False
```
Comment (NullSenseStudio) [inline generator_process/registrar.py:45]: Seamless should also have its own backend check since DreamStudio does not support it.

**Review by NullSenseStudio — CHANGES_REQUESTED** [review]
Changes looking good so far. Just a few more issues I've found.

**Comment by carson-katri** [comment]
Thank you for the very thorough review!

**Pull request merged**